### PR TITLE
Additional cid cod sku currency code fix.

### PR DESCRIPTION
**Pull request by trungnguyenexp**
Note, the fix for cid and cod (pr 229) was reverted in tealium; therefore, part of this pr will reapply that fix.

### CODE REVIEW

File: src/library/simplified/extension/currency-code.js [hunk 1]
```diff
@@ -65,6 +65,9 @@ else if (utag.isRailSearchResults() && b['entity.railSearch.searchResults.railLe
 else if (utag.isRailRateDetails() && b['entity.railSearch.railDetail.railLegs.0.railOfferItems.0.price.currency'] != undefined) {
     b['currencyCode'] = b['entity.railSearch.railDetail.railLegs.0.railOfferItems.0.price.currency'];
 }
+else if (utag.isRCO() || (b.utagPageName && b.utagPageName == "page.Checkout.Confirmation" && b.entity.productTypes && b.entity.productTypes.length == 1 && b.entity.productTypes[0] == "RAIL")){
```
Comment (demadison) [inline src/library/simplified/extension/currency-code.js]: It seems that we are checking that is the rails confirmation twice. Do we need the right side of the or condition?

File: src/library/simplified/extension/currency-code.js [hunk 2]
```diff
@@ -65,6 +65,9 @@ else if (utag.isRailSearchResults() && b['entity.railSearch.searchResults.railLe
...
+    b['currencyCode'] = b.entity.checkout.currency ? b.entity.checkout.currency : "";
```
Comment (demadison) [inline src/library/simplified/extension/currency-code.js]: Was there recent work to set the currencyCode on other confirmation pages? Should we have a single condition to set the currency code on all confirmation pages?

File: src/library/simplified/extension/currency-code.js (hunk 1 above)
Comment (trungnguyenexp) [reply src/library/simplified/extension/currency-code.js]: You are correct, I updated isRCO with the same check so will remove the rt side.

File: src/library/simplified/extension/currency-code.js (hunk 2 above)
Comment (trungnguyenexp) [reply src/library/simplified/extension/currency-code.js]: it was for this same extension, and RCO was missed.

Comment (trungnguyenexp) [reply src/library/simplified/extension/currency-code.js]: I think you are correct, this entire extension can be replaced with just the following:

if(b.pageInfo.pageName && b.pageInfo.pageName.indexOf("Checkout.Confirmation") > -1 && b.entity.checkout.currency){
    b['currencyCode'] = b.entity.checkout.currency;
}

File: src/library/simplified/extension/currency-code.js [hunk 3]
```diff
@@ -1,96 +1,6 @@
...
-    b["currencyCode"]  = b['entity.railSearch.searchResults.railLegs.0.railOfferItems.0.price.currency'];
-}
-
-else if (utag.isRailRateDetails() && b['entity.railSearch.railDetail.railLegs.0.railOfferItems.0.price.currency'] != undefined) {
-    b['currencyCode'] = b['entity.railSearch.railDetail.railLegs.0.railOfferItems.0.price.currency'];
-}
-
-//Common data element
-else if((utag.isCarPymt() || utag.isCarCO() || utag.isHCO() || utag.isHPymt()
-    || utag.isCruisePymt() || utag.isCruiseCO() || utag.isLXCO() || utag.isPCO() || utag.isPCO_Mobile() || utag.isMCO()) && typeof b["entity.checkout.cartTotal.currency"] != "undefined"){
```
Comment (demadison) [inline src/library/simplified/extension/currency-code.js]: This might be the place to add your utag.isRCO

**Comment by wmoussa** [comment]
@trungguyen Id like you to find a unit test framework for our JS code. Should we also install eslint or a JS static code analysis? I don't want to rush fixes.

@wmoussa will do. Will create a card for it and will work on it. In the mean time, do you want to put a hold on this PR? I think we need to get it out due to the checkInDate/checkOutDate fix that was reverted in Tealium on 10/12/2017.

File: src/library/simplified/extension/currency-code.js [hunk 4]
```diff
@@ -1,96 +1,6 @@
...
-    b['currencyCode'] = b['entity.packageSearch.results.offers.0.packagePrice.packageTotalPrice.currency'];
-}
-else if(utag.isLXGT_CO() && b.entity.checkout.cartTotal.currency){
-    b['currencyCode'] = b.entity.checkout.cartTotal.currency;
-}
-
-//All Slim confirmation pages
-if(b.pageInfo.pageName && b.pageInfo.pageName.indexOf("Checkout.Confirmation.Slim") > -1 && b.entity.checkout.currency){
+//All confirmation pages including slim.
+if(b.pageInfo.pageName && b.pageInfo.pageName.indexOf("Checkout.Confirmation") > -1 && b.entity.checkout.currency){
```
Comment (demadison) [inline src/library/simplified/extension/currency-code.js]: Will this new line set currency for all other pages such as HSR, HIS ?

Comment (trungnguyenexp) [reply src/library/simplified/extension/currency-code.js]: Only pages that contains "Checkout.Confirmation" such as:
page.Checkout.Confirmation
page.Flight.Checkout.Confirmation
page.Hotel.Checkout.Confirmation
page.Car.Checkout.Confirmation
page.LX.Checkout.Confirmation
etc...

File: src/library/simplified/extension/currency-code.js (hunk 3 above)
Comment (trungnguyenexp) [reply src/library/simplified/extension/currency-code.js]: yeah, this is actually the right place to add it.

File: src/library/simplified/extension/currency-code.js [hunk 5]
```diff
@@ -91,6 +91,7 @@ else if(utag.isLXGT_CO() && b.entity.checkout.cartTotal.currency){
 }
 
 //All Slim confirmation pages
```
Comment (demadison) [inline src/library/simplified/extension/currency-code.js]: should we remove slim from the comment as well ?

File: src/library/simplified/extension/currency-code.js [hunk 6]
```diff
@@ -90,7 +90,8 @@ else if(utag.isLXGT_CO() && b.entity.checkout.cartTotal.currency){
     b['currencyCode'] = b.entity.checkout.cartTotal.currency;
 }
 
-//All Slim confirmation pages
-if(b.pageInfo.pageName && b.pageInfo.pageName.indexOf("Checkout.Confirmation.Slim") > -1 && b.entity.checkout.currency){
+//All confirmation pages in cluding Slim confirmation pages.
+if(b.pageInfo.pageName && b.pageInfo.pageName.indexOf("Checkout.Confirmation") > -1 && b.entity.checkout.currency){
```
Comment (wmoussa) [inline src/library/simplified/extension/currency-code.js:94]: why all confirmation pages? I see many specific logic for different LOB: isLXCO, isCarCO, isLXGT_CO

Comment (trungnguyenexp) [reply src/library/simplified/extension/currency-code.js:94]: Not all cko pages have b.entity.checkout.currency, but most of the new uCKO pages have b.entity.checkout.currency.

File: src/library/simplified/extension/page-name-javascript-functions.js [hunk 7]
```diff
@@ -256,8 +256,10 @@ window.utag.isCruiseCO = function(){
 
 //MultiItem
 window.utag.isMCO = function(){
-	if((pageName.indexOf("page.Checkout.Confirmation") > -1 && b.entity.productTypes.length > 1 && b.entity.checkout.mickoTypeString && b.entity.checkout.mickoTypeString !== "fh")
-		|| pageName.indexOf("page.MultiItem.Checkout.Confirmation") > -1 ){
+    if(pageName.indexOf("page.MultiItem.Checkout.Confirmation") > -1 ||
+        (pageName.indexOf("page.Checkout.Confirmation") > -1
+        && b.entity.checkout.lobType && b.entity.checkout.lobType == "MultiItem"
```
Comment (tyang-expedia) [inline src/library/simplified/extension/page-name-javascript-functions.js:261]: Why don't we need the mickoTypeString == 'fh' anymore?

File: src/library/simplified/extension/currency-code.js (hunk 6 above)
Comment (wmoussa) [reply src/library/simplified/extension/currency-code.js:94]: Could we use ‘else if’ at least? Because looks like this if will always get evaluated

**Pull request merged**